modules/vmseries/main.tf

```diff
@@ -113,13 +113,24 @@ resource "azurerm_virtual_machine" "this" {
     identity_ids = var.identity_ids
   }
 }
+resource "azurerm_log_analytics_workspace" "this" {
+  count = var.metrics_retention_in_days != 0 && var.application_insights_mode_workspace ? 1 : 0
```

Comment:
This is just an enhancement idea, but I thought that maybe we could change that creation condition based on retention days count to some `true/false` flag? Like we have in different modules. 
Something like:
```
count = var.create_appinsights && var.application_insights_mode_workspace ? 1 : 0
```

with

```
variable "create_appinsights" {
  default = true
}
```

Being dependent only on number of retention days saves us one variable, but IMO it's not that obvious when using the module.

Reply:
Agreed, I also thought about suggesting such an approach.

Additionally, apart from the conditional var, all other AI/log analytics/metrics variables could potentially be wrapped into a single map.